src/pages/SwapList/LedgerActionDialogBody.tsx

```diff
@@ -140,34 +176,59 @@ function LedgerActionDialogBody({
             {actionDoneBefore && <ActionDoneMessage />}
             <Typography paragraph={true}>
               Deploy the following contract to the Ethereum{" "}
-              <b>{action.payload.network}</b> network with <b>{amount}</b> ETH:
+              <b>{action.payload.network}</b> network with{" "}
+              <b>{mainUnitAmount}</b> ETH using a gas limit of <b>{gasLimit}</b>{" "}
+              wei:
             </Typography>
             <Typography
               variant="body1"
               style={{
                 wordWrap: "break-word"
               }}
             >
-              {action.payload.data}
+              {contract}
             </Typography>
           </DialogContent>
+          {!web3 && (
+            <DialogActions>
+              <CopyToClipboardButton content={mainUnitAmount} name="amount" />
+              <CopyToClipboardButton content={contract} name="contract code" />
+              <CopyToClipboardButton
+                content={gasLimit.toString()}
+                name="gas limit"
+              />
+            </DialogActions>
+          )}
           <DialogActions>
             <Button onClick={onClose} variant="contained" color="secondary">
               Cancel
             </Button>
+            {!web3 && (
+              <Button
+                onClick={onManualConfirmation}
+                variant="contained"
+                color="primary"
+              >
+                Confirm
+              </Button>
+            )}
             <Web3SendTransactionButton
               transactionConfig={{
-                data: action.payload.data,
+                data: contract,
                 value: action.payload.amount,
-                gas: +action.payload.gas_limit
+                gas: gasLimit
               }}
-              onSuccess={onSuccessEthereum}
+              onSuccess={onSuccessMetaMask}
             />
           </DialogActions>
         </React.Fragment>
       );
     }
     case "ethereum-call-contract": {
+      const address = action.payload.contract_address;
+      const data = action.payload.data;
+      const gasLimit = parseInt(action.payload.gas_limit, 16);
```

Comment:
❓ `gasLimit` and `data` (above `contract`) are the same in the `ethereum-deploy-contract` case - would it make sense to pull them out and reuse them?